### PR TITLE
Add ReadTask implementation

### LINKED ISSUE #4806: [BPS] New task/activity to read resources and populate variables
**Description:**
ReadTask is a custom BPMN Activity that is capable of reading resources from a source and populate it to given variable (by name)

 Supported sources:
      1. REGISTRY : WSO2 carbon registry
      2. ENVIRONMENT : Environment variables
      3. SYSTEM : Java system variables

```xml
 <serviceTask xmlns:activiti="http://activiti.org/bpmn" id="servicetask" name="Service Task" activiti:class="org.wso2.carbon.bpmn.extensions.tasks.ReadTask">
    <extensionElements>
       <activiti:field name="resource">
          <activiti:expression><![CDATA[conf:/custom/testJson]]></activiti:expression>
       </activiti:field>
       <activiti:field name="origin">
          <activiti:expression><![CDATA[REGISTRY]]></activiti:expression>
       </activiti:field>
       <activiti:field name="target">
          <activiti:expression><![CDATA[testRegJsonVar]]></activiti:expression>
       </activiti:field>
       <activiti:field name="type">
          <activiti:expression><![CDATA[json]]></activiti:expression>
       </activiti:field>
    </extensionElements>
 </serviceTask>
```

1. resource (mandatory) - resource name (incase if the resource is registry resource, this should be resource path)
2. origin (mandatory) - origin / source type. Supported sources : REGISTRY, ENVIRONMENT, SYSTEM
3. target (mandatory) - target variable name that should be populated with the content read by this task
4. type (optional) - type of the variable name. Content read by this task will be parsed / converted to given type. Supported types: string, integer, boolean, json, xml

### PR DESCRIPTION
**Pull request by milindaperera**
Fix: https://github.com/wso2/product-ei/issues/4806

### CODE REVIEW

File: components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            }
+
+        } else if (ORIGIN_TYPE_ENV.equalsIgnoreCase(getExpressionStrValue(origin, execution))) {
+
+            execution.setVariable(targetVarName, typeConvert(System.getenv(resourceName), targetType));
+
+        } else if (ORIGIN_TYPE_SYS.equalsIgnoreCase(getExpressionStrValue(origin, execution))) {
+
+            execution.setVariable(targetVarName, typeConvert(System.getProperty(resourceName), targetType));
+
```
Comment (sameeragunarathne) [inline components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java]: extra new line?

File: components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    private Expression origin; // Source/Origin of the resource
+    private Expression resource; // Resource name
+    private Expression target; // Target variable name
+    private Expression type; // data type of the resource. Default is string
+
+    @Override
+    public void execute(DelegateExecution execution) throws Exception {
+
+        String originStr = getExpressionStrValue(origin, execution);
+        if (originStr == null) {
```
Comment (sameeragunarathne) [inline components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java]: can this be empty?

File: components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            throw new WSO2TaskException("\"origin\" parameter is mandatory, but found null");
+        }
+
+        String resourceName = getExpressionStrValue(resource, execution);
+        if (resourceName == null) {
```
Comment (sameeragunarathne) [inline components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java]: same as https://github.com/wso2/carbon-business-process/pull/611/files#r350924399

File: components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java (hunk 1 above)
Comment (milindaperera) [reply components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java]: Fixed, Thanks

File: components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java (hunk 2 above)
Comment (milindaperera) [reply components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java]: Fixed, Thanks

File: components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java (hunk 3 above)
Comment (milindaperera) [reply components/bpmn/org.wso2.carbon.bpmn.extensions/src/main/java/org/wso2/carbon/bpmn/extensions/tasks/ReadTask.java]: Fixed, Thanks

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**